[Exec] move script execution logic to query package

ramtinms opened the pull request.

ramtinms commented on engine/execution/computation/manager.go:
```diff
@@ -87,21 +75,15 @@ type ComputationConfig struct {
-	log                      zerolog.Logger
-	tracer                   module.Tracer
-	metrics                  module.ExecutionMetrics
-	me                       module.Local
```
`me` was a legacy thing, not used anywhere.

ramtinms commented on engine/execution/computation/manager.go:
```diff
@@ -87,21 +75,15 @@ type ComputationConfig struct {
 // Manager manages computation and execution
 type Manager struct {
-	log                      zerolog.Logger
-	tracer                   module.Tracer
```
tracer is moved to computer and query modules.

pattyshack commented on engine/execution/computation/manager.go:
```diff
@@ -71,11 +60,10 @@ type ComputationManager interface {
-	DerivedDataCacheSize     uint
-	ScriptLogThreshold       time.Duration
-	ScriptExecutionTimeLimit time.Duration
+	QueryConfig          query.Config
```
if you rename query.Config to query.QueryConfig, you can use anonymous composition / reduce the number of changes

pattyshack commented:
maybe split this into separate PRs: actual code change vs moving the code

ramtinms commented on engine/execution/computation/query/executor.go:
```diff
@@ -0,0 +1,238 @@
+	return encodedValue, nil
+}
+
+func summarizeLog(log string, limit int) string {
+	if limit > 0 && len(log) > limit {
+		split := int(limit/2) - 1
+		var sb strings.Builder
+		sb.WriteString(log[:split])
+		sb.WriteString(" ... ")
+		sb.WriteString(log[len(log)-split:])
+		return sb.String()
+	}
+	return log
+}
```
This functionality has been moved outside of ExecuteScript for better clarity, it has also changed to handle the case the limit is set to zero.

ramtinms commented on engine/execution/computation/manager.go:
```diff
@@ -25,16 +19,10 @@ import (
 )
 
 const (
-	DefaultScriptLogThreshold       = 1 * time.Second
-	DefaultScriptExecutionTimeLimit = 10 * time.Second
-
-	MaxScriptErrorMessageSize = 1000 // 1000 chars
-
```
Moved to query package

ramtinms commented on engine/execution/computation/manager.go:
```diff
@@ -71,11 +60,10 @@ type ComputationManager interface {
-	CadenceTracing           bool
-	ExtensiveTracing         bool
-	DerivedDataCacheSize     uint
-	ScriptLogThreshold       time.Duration
-	ScriptExecutionTimeLimit time.Duration
```
Moved to queryConfigs

ramtinms commented:
@pattyshack not that much code change, I reverted some of the changes for this PR and added comments for minor ones, 95% of this code should be just moving things around, I'll make another PR with actual important changes

zhangchiqing commented on engine/execution/computation/manager.go:
```diff
@@ -87,21 +75,15 @@ type ComputationConfig struct {
+	derivedChainData *derived.DerivedChainData
 }
 
+var _ ComputationManager = &Manager{}
```
```suggestion
var _ ComputationManager = (*Manager)(nil)
```

pattyshack reviewed (approved):
could you click on the re-review icon next time to catch my attention?  I usually ignore the email updates.
